diff --git a/dockers/docker-base/Dockerfile.j2 b/dockers/docker-base/Dockerfile.j2
index 45e91fbab..905c59e34 100644
--- a/dockers/docker-base/Dockerfile.j2
+++ b/dockers/docker-base/Dockerfile.j2
@@ -27,6 +27,8 @@ RUN rm -rf               \
 # Make apt-get non-interactive
 ENV DEBIAN_FRONTEND=noninteractive

+RUN apt-get update && apt-get install -y ca-certificates
+
 # Configure data sources for apt/dpkg
 COPY ["dpkg_01_drop", "/etc/dpkg/dpkg.cfg.d/01_drop"]
 {% if CONFIGURED_ARCH == "armhf" %}
diff --git a/onie-image.conf b/onie-image.conf
index b0e2796a8..97f2e57a4 100644
--- a/onie-image.conf
+++ b/onie-image.conf
@@ -37,7 +37,7 @@ OUTPUT_ONIE_IMAGE=target/sonic-$TARGET_MACHINE.bin
 OUTPUT_RAW_IMAGE=target/sonic-$TARGET_MACHINE.raw

 ## Raw image size in MB
-RAW_IMAGE_DISK_SIZE=1700
+RAW_IMAGE_DISK_SIZE=32768

 ## Output file name for kvm image
 OUTPUT_KVM_IMAGE=target/sonic-$TARGET_MACHINE.img
diff --git a/sonic-slave-bullseye/Dockerfile.j2 b/sonic-slave-bullseye/Dockerfile.j2
index 7bcf438cc..d728a2399 100644
--- a/sonic-slave-bullseye/Dockerfile.j2
+++ b/sonic-slave-bullseye/Dockerfile.j2
@@ -21,6 +21,8 @@ FROM {{ prefix }}debian:bullseye

 MAINTAINER gulv@microsoft.com

+RUN apt-get update && apt-get install -y ca-certificates
+
 COPY ["no-check-valid-until", "/etc/apt/apt.conf.d/"]
 COPY ["apt-retries-count", "/etc/apt/apt.conf.d"]

diff --git a/sonic-slave-buster/Dockerfile.j2 b/sonic-slave-buster/Dockerfile.j2
index bb1ec2c97..37141e34b 100644
--- a/sonic-slave-buster/Dockerfile.j2
+++ b/sonic-slave-buster/Dockerfile.j2
@@ -23,6 +23,8 @@ MAINTAINER gulv@microsoft.com
 COPY ["no-check-valid-until", "/etc/apt/apt.conf.d/"]
 COPY ["apt-retries-count", "/etc/apt/apt.conf.d"]

+RUN apt-get update && apt-get install -y ca-certificates
+
 {%- if CROSS_BUILD_ENVIRON != "y" %}
 COPY ["sources.list.{{ CONFIGURED_ARCH }}", "/etc/apt/sources.list"]
 {%- else %}
diff --git a/sonic-slave-jessie/Dockerfile.j2 b/sonic-slave-jessie/Dockerfile.j2
index 96ac4653e..2357fd098 100644
--- a/sonic-slave-jessie/Dockerfile.j2
+++ b/sonic-slave-jessie/Dockerfile.j2
@@ -9,6 +9,8 @@ FROM {{ prefix }}debian:jessie

 MAINTAINER johnar@microsoft.com

+RUN apt-get update && apt-get install -y ca-certificates
+
 COPY ["no-check-valid-until", "/etc/apt/apt.conf.d/"]
 COPY ["apt-retries-count", "/etc/apt/apt.conf.d"]

diff --git a/sonic-slave-stretch/Dockerfile.j2 b/sonic-slave-stretch/Dockerfile.j2
index 5d2d345e2..55334ff8f 100644
--- a/sonic-slave-stretch/Dockerfile.j2
+++ b/sonic-slave-stretch/Dockerfile.j2
@@ -9,6 +9,8 @@ FROM {{ prefix }}debian:stretch

 MAINTAINER gulv@microsoft.com

+RUN apt-get update && apt-get install -y ca-certificates
+
 COPY ["no-check-valid-until", "/etc/apt/apt.conf.d/"]
 COPY ["apt-retries-count", "/etc/apt/apt.conf.d"]
 COPY ["sources.list.{{ CONFIGURED_ARCH }}", "/etc/apt/sources.list"]
