diff --git a/sonic-mgmt/tests/common/plugins/sanity_check/checks.py b/sonic-mgmt/tests/common/plugins/sanity_check/checks.py
--- a/sonic-mgmt/tests/common/plugins/sanity_check/checks.py
+++ b/sonic-mgmt/tests/common/plugins/sanity_check/checks.py
@@ -40,6 +40,4 @@
     'check_mux_simulator',
-    'check_orchagent_usage',
-    'check_bfd_up_count',
-    'check_mac_entry_count']
+    'check_orchagent_usage']
 
 __all__ = CHECK_ITEMS
 
