diff --git a/src/core/lib/iomgr/load_file.cc b/src/core/lib/iomgr/load_file.cc
index 9068670118..a4d9bc95b2 100644
--- a/src/core/lib/iomgr/load_file.cc
+++ b/src/core/lib/iomgr/load_file.cc
@@ -55,7 +55,6 @@ grpc_error_handle grpc_load_file(const char* filename, int add_null_terminator,
   if (bytes_read < contents_size) {
     gpr_free(contents);
     error = GRPC_OS_ERROR(errno, "fread");
-    GPR_ASSERT(ferror(file));
     goto end;
   }
   if (add_null_terminator) {
